scripts-available/CDB_FederatedServerDiagnostics.sql

```diff
@@ -81,6 +81,22 @@ $$
 LANGUAGE PLPGSQL VOLATILE PARALLEL UNSAFE;
 
 
+--
+-- Get the foreign server options of a remote PG server
+--
+CREATE OR REPLACE FUNCTION @extschema@.__CDB_FS_Foreign_Server_Options_PG(server_internal name)
+RETURNS jsonb
+AS $$
+    -- See https://www.postgresql.org/docs/current/catalog-pg-foreign-server.html
+    SELECT jsonb_object(server_options) FROM (
```

Comment:
2 things:
- I'm not sure what's all the unnest + regex about. Could it be solved using `pg_options_to_table`?
- This might return the remote password, which we should avoid if we don't want to worry about it later (caching, logs, etc).

Reply:
> I'm not sure what's all the unnest + regex about. Could it be solved using pg_options_to_table?

this is about converting `srvoptions` from a `Foreign server specific options, as “keyword=value” strings` into a json object. I did not know about `pg_options_to_table` and I'm not sure it can be applied there, but if I can simplify it with it or some other function, I'll do.

> This might return the remote password, which we should avoid if we don't want to worry about it later (caching, logs, etc).

Ok.

Reply:
PS: the password comes in the user mapping, so that's one less concern to worry about.

Reply:
> PS: the password comes in the user mapping, so that's one less concern to worry about.

That's true. Sorry for the noise.